base/loading.jl

```diff
@@ -1926,12 +1926,7 @@ function precompile(@nospecialize(f), args::Tuple)
     precompile(Tuple{Core.Typeof(f), args...})
 end
 
-function precompile(argt::Type)
-    if ccall(:jl_compile_hint, Int32, (Any,), argt) == 0
-        @warn "Inactive precompile statement" maxlog=100 form=argt _module=nothing _file=nothing _line=0
-    end
-    true
-end
+precompile(argt::Type) = ccall(:jl_compile_hint, Int32, (Any,), argt) != 0
```

Comment:
```suggestion
const ENABLE_PRECOMPILE_WARNINGS = Ref(false)
function precompile(argt::Type)
    ret = ccall(:jl_compile_hint, Int32, (Any,), argt) != 0
    if !ret && ENABLE_PRECOMPILE_WARNINGS[]
        @warn "Inactive precompile statement" maxlog=100 form=argt _module=nothing _file=nothing _line=0
    end
    return ret
end
```
Presumably we'd also want to add a way to set that const via `Pkg.test`.

Reply:
For interactive work or? For normal packages it will be hard to set this variable since the precompile process is not the same as the parent process.

Reply:
I'm fine with that too, but a global setting rather than something that has to be passed with each `precompile` call does seem to have some ergonomic advantages.